cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/log/LogStageAppender.java

```diff
@@ -46,6 +46,19 @@ protected void append(ILoggingEvent eventObject) {
     }
   }
 
+  @Override
+  public void stop() {
+    super.stop();
+    for (Appender appender : appenders) {
+      try {
+        appender.stop();
+      } catch (Throwable t) {
+        // catch this so that we keep going and try to stop all appenders
```

Comment:
Would it be better to do something like this: https://github.com/caskdata/cdap/blob/develop/cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/MultipleOutputs.java#L295

(throw an exception at the end)